sdcm/remote/libssh2_client/__init__.py

```diff
@@ -0,0 +1,745 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+# pylint: disable=duplicate-code
+from typing import List, Union, Iterable, Optional, Dict
+from dataclasses import dataclass, field
+from time import perf_counter, sleep
+from os.path import normpath, expanduser, exists
+from sys import float_info
+from io import StringIO
+from warnings import warn
+from socket import socket, AF_INET, SOCK_STREAM, gaierror, error as sock_error
+from select import select
+from threading import Thread, Lock
+from multiprocessing import Queue, BoundedSemaphore, Event
+from ssh2.channel import Channel  # pylint: disable=no-name-in-module
+from ssh2.exceptions import SocketRecvError, AuthenticationError  # pylint: disable=no-name-in-module
+from ssh2.session import Session as LibSSH2Session, LIBSSH2_SESSION_BLOCK_INBOUND, LIBSSH2_SESSION_BLOCK_OUTBOUND  # pylint: disable=no-name-in-module
+from ssh2.error_codes import LIBSSH2_ERROR_EAGAIN  # pylint: disable=no-name-in-module
+from .exceptions import AuthenticationException, UnknownHostException, ConnectError, PKeyFileError, UnexpectedExit, \
+    CommandTimedOut, FailedToReadCommandOutput, ConnectTimeout, FailedToRunCommand, OpenChannelTimeout
+from .result import Result
+
+
+__all__ = ['Session', 'Timings', 'Client', 'Channel', 'FailedToRunCommand']
+
+
+LINESEP = b'\n'
+
+
+NullableTiming = Optional[float]
+Timing = float
+Delays = Iterable[float]
+
+
+class __DEFAULT__:  # pylint: disable=invalid-name, too-few-public-methods
+    """ Default value for function attribute when None is not an option """
+
+
+class Session(LibSSH2Session):  # pylint: disable=too-few-public-methods
+    """Custom SSH2 Session class with Lock in it, to make it thread safe where it is needed """
+
+    def __init__(self):
+        # A lock that is used to make it thread safe
+        self.lock = Lock()
+        super(Session, self).__init__()
```

Comment:
It'll be nice to make it more Python3-ish and use just `super()`

Reply:
fixed